Marlin/src/module/stepper/cycles.h

```diff
@@ -126,20 +126,20 @@ constexpr uint32_t min_isr_loop_cycles = (isr_mixing_stepper_cycles LOGICAL_AXIS
 #endif
 
 // Calculate the minimum MPU cycles needed per pulse to enforce
-constexpr uint32_t min_stepper_pulse_cycles = _min_pulse_high_ns * CYCLES_PER_MICROSECOND / 1000;
+constexpr hal_timer_t min_stepper_pulse_cycles = _min_pulse_high_ns * CYCLES_PER_MICROSECOND / 1000;
```

Comment:
This commit is problematic. `MIN_STEPPER_PULSE_CYCLES` is definitely `uint32_t` (cycles) and not `hal_timer_t` i.e. timer ticks (usually cycles/4).
I think basically all of the changed lines in this file are cycles and not `hal_timer_t`.

Reply:
Please point out any others that should be `uint32_t` which were missed. As I mentioned in another comment, we will surely need 32-bits for any timer values that are being multiplied or shifted so they don't overflow on AVR. We can also use a minimum size, which would make `hal_timer_t` 32-bits on all platforms where it isn't 64 bits.